File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/EncryptionKey.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "encryption_keys", schema = "public")
public class EncryptionKey {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "encryption_key_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "encryption_key_private_key", nullable = false, length = Integer.MAX_VALUE)
    private String encryptionKeyPrivateKey;

    @NotNull
    @Column(name = "encryption_key_public_key", nullable = false, length = Integer.MAX_VALUE)
    private String encryptionKeyPublicKey;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

}

File: umbrella-insurance-react-ui/src/models/users/encryptedKeys/v1/EncryptionKey.ts
import { User } from "../../v1/User";

export class EncryptionKey {
    id?: number;
    user?: User;
    encryptionKeyPrivateKey?: string;
    encryptionKeyPublicKey?: string;
    createdDateTime?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.user) {
                this.user = new User(obj.user);
            }
            if(obj?.encryptionKeyPrivateKey) {
                this.encryptionKeyPrivateKey = obj.encryptionKeyPrivateKey;
            }
            if(obj?.encryptionKeyPublicKey) {
                this.encryptionKeyPublicKey = obj.encryptionKeyPublicKey;
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
        }
    }
}